### DISCUSSION TITLE
3D axisymmetry to 2D

### DISCUSSION THREAD

**Discussion by sidharthsarmah** [idaholab/moose#25671] [Q&A Meshing]
Hi @GiudGiud ,

<img width="355" alt="image" src="https://github.com/idaholab/moose/assets/110927986/fb08bf1d-6a6b-4a8e-afe8-642ab7a0e923">

When I wished to do this extruded 3D strutted in 2D, I used:

```
[Problem]
  type = FEProblem
  coord_type = XYZ
[]
```
<img width="662" alt="image" src="https://github.com/idaholab/moose/assets/110927986/e075df70-c968-4ab1-a4fe-e55ace046234">

Could you suggest, what should be it for an axisymmetry case, I don't want an extruded one like in the figure as both have similar projection. I want the one on the right. Does this work? Thanks.

```
[Problem]
  type = FEProblem
  coord_type = RZ
[]
```

**Comment by GiudGiud** [comment]
you specify the symmetry axis (in the Mesh, in the Problem it is a deprecated parameter)

pick the axis that is horizontal in the mesh you show at the bottom

**Reply by sidharthsarmah** [reply]
I did not get it, I used a mesh made in ABAQUS, which has coords for nodes and elements? How do I specify the axis? 
In this case, it is :
<img width="474" alt="image" src="https://github.com/idaholab/moose/assets/110927986/e103dbfd-5f1c-4cda-bc7e-cd0df17da2b6">

**Reply by GiudGiud** [reply]
in the [Mesh] block in the input there is a symmetry axis parameter to set

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud ,

I just used this as it is mentioned here:
```
If the RZ coordinate system is used, there are two options for how to specify the coordinate axis(es) in an input file:
Specify ["rz_coord_axis"](https://mooseframework.inl.gov/syntax/Mesh/#2ce1a211-2d33-4595-9032-b4227db24ba2) to choose a single RZ coordinate system, using the  direction and starting at  (0,0,0). If the former is used, then the axial coordinate is x, and the radial coordinate is y
 ```
https://mooseframework.inl.gov/syntax/Mesh/

```
  [./mesh]
    type = FileMeshGenerator
    file = Cyl.inp
    coord_type = RZ
    rz_coord_axis = X
  [../]
```
```

*** ERROR ***
/Users/sidharthsarmah/MOOSE/projects/moose/modules/tensor_mechanics/CZM/Cyl/cyl.i:17: unused parameter 'Mesh/mesh/coord_type'
/Users/sidharthsarmah/MOOSE/projects/moose/modules/tensor_mechanics/CZM/Cyl/cyl.i:18: unused parameter 'Mesh/mesh/rz_coord_axis'
```

**Reply by GiudGiud** [reply]
```
[Mesh]
  coord_type = RZ
  rz_coord_axis = X
  [./mesh]
    type = FileMeshGenerator
    file = Cyl.inp
  [../]
[]
```

**Reply by sidharthsarmah** [reply]
Thanks

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , I have one confusion, when I apply a point load in 2D, it resembles a line load in 3D like this:
<img width="354" alt="image" src="https://github.com/idaholab/moose/assets/110927986/60e13057-27b2-4f5c-830d-6b59b404d1fd">

When I apply a line load in the RZ coord, it seems the load in something like this (right), but I want it to be like the other way(left), a load on top for bending:
<img width="459" alt="image" src="https://github.com/idaholab/moose/assets/110927986/d357cf6d-f644-4896-b20f-17345ade3910">
```
[Mesh]
  coord_type = RZ
  rz_coord_axis = X
  [./mesh]
    type = FileMeshGenerator
    file = Cyl.inp
  [../]
  [./bounding_box]
    type = BoundingBoxNodeSetGenerator
    input = 'mesh'
    new_boundary = 'nodes_to_convert_to_sides'
    top_right = '25.3999996  5.71500015 0'
    bottom_left = '20 5 0'
  [../]
  [./create_sideset]
    type = SideSetsFromNodeSetsGenerator
    input = 'bounding_box'
  [../]
[]
```
```
[BCs]
  [./u_top_pull]
    type = ADPressure
    variable = disp_y
    boundary = load
    function = 0.1*t
  []
```

**Reply by GiudGiud** [reply]
This is all expected, and there's no way around it. It's by definition of these coordinate systems.

**Reply by sidharthsarmah** [reply]
Hi @GiudGiud , I agree on this but is there any way to just add a load at a particular theta along the radius? That would be interesting.

**Reply by GiudGiud** [reply]
nope we unfortunatly do not have R-theta-z implemented, only RZ
